### PR TITLE
Added more javadocs

### PR DESCRIPTION
**Pull request by 0utplay**
This pull request includes:

- [ ] breaking changes
- [x] no breaking changes

### Changes made to the repository:

Added some javadocs, updated some examples and fixed some typos in already existing docs

### Documentation of test results:

Recompiling the cloud worked without any issues

### CODE REVIEW

File: cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionGroup.java [hunk 1]
```diff
@@ -19,32 +19,91 @@
...
 
+  /**
+   * Sets the color of this group
+   *
+   * @param color the color to set
+   */
   void setColor(@NotNull String color);
 
+
   String getSuffix();
```
Comment (derklaro) [inline cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionGroup.java:89]: I guess you missed that

File: cloudnet-examples/src/main/java/de/dytanic/cloudnet/examples/sign/ExampleSigns.java [hunk 2]
```diff
@@ -16,23 +16,38 @@
...
 import org.bukkit.event.EventHandler;
+import org.jetbrains.annotations.NotNull;
 
 public final class ExampleSigns {
 
   // getting the SignManagement via CloudNet's service registry
-  private final AbstractSignManagement signManagement = CloudNetDriver.getInstance().getServicesRegistry()
-    .getFirstService(AbstractSignManagement.class);
+  private final SignManagement signManagement = CloudNetDriver.getInstance().getServicesRegistry()
+    .getFirstService(SignManagement.class);
```
Comment (derklaro) [inline cloudnet-examples/src/main/java/de/dytanic/cloudnet/examples/sign/ExampleSigns.java:40]: Is this change possible?

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/player/ICloudOfflinePlayer.java [hunk 3]
```diff
@@ -43,7 +43,7 @@
   void setName(@NotNull String name);
 
   /**
-   * Returns the XBoxId from the offlinePlayer if the player has to be connect from the Minecraft Bedrock Edition
+   * Returns the XBoxId from the offlinePlayer if the player ever connected using the Minecraft Bedrock Edition
    */
   String getXBoxId();
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/player/ICloudOfflinePlayer.java:50]: I guess we should mark this nullable. Also please use @return here

File: cloudnet/src/main/java/de/dytanic/cloudnet/CloudNet.java [hunk 4]
```diff
@@ -171,12 +171,14 @@
...
+/**
+ * Represents the implementation of the {@link CloudNetDriver} for nodes.
+ */
 public final class CloudNet extends CloudNetDriver {
 
   public static final int TPS = 10;
-  private static CloudNet instance;
   private static final Logger LOGGER = LogManager.getLogger(CloudNet.class);
-
+  private static CloudNet instance;
```
Comment (derklaro) [inline cloudnet/src/main/java/de/dytanic/cloudnet/CloudNet.java:182]: I think we should keep a little bit of formatting here

File: cloudnet/src/main/java/de/dytanic/cloudnet/util/JavaVersionResolver.java [hunk 5]
```diff
@@ -35,6 +38,12 @@ private JavaVersionResolver() {
     throw new UnsupportedOperationException();
   }
 
+  /**
+   * Starts a process and checks the java version of the given input
+   *
+   * @param input the path to the java executable to check the version for
+   * @return the java version of the executable, null if the input is no valid path to an executable
+   */
   public static JavaVersion resolveFromJavaExecutable(String input) {
```
Comment (derklaro) [inline cloudnet/src/main/java/de/dytanic/cloudnet/util/JavaVersionResolver.java:49]: Please mark this nullable too

**Review by derklaro — CHANGES_REQUESTED** [review]

File: cloudnet-examples/src/main/java/de/dytanic/cloudnet/examples/sign/ExampleSigns.java (hunk 2 above)
Comment (0utplay) [reply cloudnet-examples/src/main/java/de/dytanic/cloudnet/examples/sign/ExampleSigns.java:40]: Why should it not be possible?

Comment (derklaro) [reply cloudnet-examples/src/main/java/de/dytanic/cloudnet/examples/sign/ExampleSigns.java:40]: See https://github.com/CloudNetService/CloudNet-v3/blob/development/cloudnet-common/src/main/java/de/dytanic/cloudnet/common/registry/DefaultServicesRegistry.java#L47 

The class which get's registered is in the backing map (in this case AbstractSignManagement), but you try to read a SignManagement. This behaviour should be changed in the future by rewriting the lookup in the registry but for now this change will not work.

Comment (0utplay) [reply cloudnet-examples/src/main/java/de/dytanic/cloudnet/examples/sign/ExampleSigns.java:40]: How does this work than: https://github.com/CloudNetService/CloudNet-v3/blob/eb8144865381261b6b5c996b8873cb908678e260/cloudnet-modules/cloudnet-signs/src/main/java/eu/cloudnetservice/cloudnet/ext/signs/bukkit/BukkitSignManagement.java#L50

Comment (derklaro) [reply cloudnet-examples/src/main/java/de/dytanic/cloudnet/examples/sign/ExampleSigns.java:40]: My mistake i was thinking about the old sign management.

**Pull request merged**